nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -1392,6 +1407,45 @@ public boolean accept(final File dir, final String filename) {
         }
     }
 
+    private String printBasicOSJavaDetails(final String xms, final String xmx) {
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode details = mapper.createObjectNode();
+
+        // java version
+        details.put("javaVersion", System.getProperty("java.version"));
+
+        // num cores
+        final OperatingSystemMXBean os = ManagementFactory.getOperatingSystemMXBean();
+        details.put("cores", os.getAvailableProcessors());
+
+        // max file descriptor count + total physical memory
+        try {
+            final OperatingSystemMXBean osStats = ManagementFactory.getOperatingSystemMXBean();
+            final Class<?> unixOsMxBeanClass = Class.forName("com.sun.management.UnixOperatingSystemMXBean");
+            if (unixOsMxBeanClass.isAssignableFrom(osStats.getClass())) {
+                final Method totalPhysicalMemory = unixOsMxBeanClass.getMethod("getTotalPhysicalMemorySize");
```

Comment:
In this case, is the total physical memory useful? The `Runtime` memory methods provide heap information, which is actually addressable for the process, versus the physical memory, which may or may not be available to the JVM.

Reply:
I believe that some users tend to set a very high heap, not leaving enough room for off-heap. Having the total memory available on the host may be useful. If the information is not available, we would see a null value.